### PR TITLE
feat: create_ publickey, sharedkey, selfkey

### LINKED ISSUE #78: at_c: atclient_atkey_create_publickey
'Constructor' for creating a Public AtKey 

Populates struct `atclient_atkey *atkey` based on other passed arguments.

Located in `atclient/src/atkey.c`

```c
int atclient_atkey_create_publickey(atclient_atkey *atkey, const char *name, const char *sharedby, const char *namespacestr)
{
    return 1; // TODO: implement
}
```

### LINKED ISSUE #79: at_c: atclient_atkey_create_selfkey
'Constructor' for populating the `atclient_atkey *atkey` with the correct variables according to the corresponding arguments passed.

Located in `atclient/src/atkey.c`

```c
int atclient_atkey_create_selfkey(atclient_atkey *atkey, const char *name, const char *sharedby, const char *namespacestr)
{
    return 1; // TODO: implement
}
```

### LINKED ISSUE #80: at_c: atclient_atkey_create_sharedkey
'Constructor' for populating the `atclient_atkey *atkey` with the corresponding arguments passed into the function that  represent a SharedKey AtKey.

Located in `atclient/src/atkey.c`

```c
int atclient_atkey_create_sharedkey(atclient_atkey *atkey, const char *name, const char *sharedby, const char *sharedwith, char *namespacestr)
{
    return 1; // TODO: implement
}
```

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
closes #78
closes #79 
closes #80 

**- What I did**

- implemented 3 functions in atkey.c 
- `atclient_atkey_create_sharedkey`
- `atclient_atkey_create_selfkey`
- `atclient_atkey_create_publickey`

- 3 unit tests for each of these in `test_atkey_create.c`
- `test_create_publickey`
- `test_create_sharedkey`
- `test_create_selfkey`

### CODE REVIEW

**Comment by xavierchanth** [comment]
@JeremyTubongbanua what needs to be done

**Comment by JeremyTubongbanua** [comment]
> @JeremyTubongbanua what needs to be done

just needed to finish up the last 2 unit tests, which have just been committed.

Ready for review

**Comment by xavierchanth** [comment]
Thank you!

File: packages/atclient/src/atkey.c [hunk 1]
```diff
@@ -217,138 +217,249 @@ int atclient_atkey_from_atstr(atclient_atkey *atkey, const atclient_atstr atstr)
...
-        goto exit;
-    }
+  int ret = 1;
 
-    ret = 0;
+  ret = atclient_atkey_to_string(atkey, atstr->str, atstr->olen, &(atstr->olen));
+  if (ret != 0) {
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atkey_to_string failed\n");
     goto exit;
+  }
```
Comment (xavierchanth) [inline packages/atclient/src/atkey.c:323]: Everything above this point is formatting

**Review by xavierchanth — APPROVED** [review]

**Pull request merged**